core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -379,6 +379,41 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
     }
   executorAllocationManager.foreach(_.start())
 
+  // Use reflection to instantiate listeners specified via the `spark.extraListeners` configuration
+  // or the SPARK_EXTRA_LISTENERS environment variable
+  try {
+    val listenerClassNames: Seq[String] = {
+      val fromSparkConf = conf.get("spark.extraListeners", "").split(',')
+      val fromEnvVar = Option(conf.getenv("SPARK_EXTRA_LISTENERS")).getOrElse("").split(',')
```

Comment:
I see. I think at some point it would be nice to have a better long-term solution for this kind of scenario. I added `SPARK_DIST_CLASSPATH` in #2982 for similar reasons, but if we have something like "spark-dist.conf" in `SPARK_CONF_DIR`, we could at some point merge those values with the user's configuration and avoid adding more env variables like this.

Anyway, separate discussion.